### PR TITLE
Add badger store for Extension

### PR DESCRIPTION
**Pull request by flaviodsr**
Implement ExtensionStore interface using Badger as storage.

### CODE REVIEW

File: pkg/core/store/badger/extension.go [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+	// TODO: Replace with a badgerhold query.
+	if query != nil {
+		if query.ExtensionID != "" {
+			fullExtension, err := es.GetExtension(ctx, query.ExtensionID)
+			if err == nil {
+				matchingExtension := fullExtension.GetExtensionIfMatch(query)
+				if matchingExtension != nil {
+					result = append(result, matchingExtension)
+				}
+			}
```
Comment (jsuchome) [inline pkg/core/store/badger/extension.go:56]: Why is the potential error ignored here?

File: pkg/core/store/badger/extension.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+// AddExtension adds a new extension to the store.
+func (es *ExtensionStore) AddExtension(ctx context.Context, extension *domain.Extension) (*domain.Extension, error) {
+	extension.EnsureID(ctx, es)
+	extension.SetCreated(ctx)
+
+	err := es.store.Insert(extension.ID, extension)
+	if err != nil {
+		return nil, domain.NewErrExtensionExists(extension.ID)
```
Comment (jsuchome) [inline pkg/core/store/badger/extension.go:28]: Just curious: can't there be a different reason the store operation fails?

This applies to all error checks after badgerstore operations

Comment (flaviodsr) [reply pkg/core/store/badger/extension.go:28]: According to the docs, insert only fails if a key already exists (see: https://github.com/timshannon/badgerhold#behavior-changes)

File: pkg/core/store/badger/extension.go (hunk 1 above)
Comment (flaviodsr) [reply pkg/core/store/badger/extension.go:56]: `GetExtension` returns an error only if the extension cannot be found and in that case this functions will return `result` (empty list). So, there is no need to check for the error as it is expected when there is no extension with the specified ID.

File: pkg/core/store/badger/extension.go [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+	newExtension.Created = extension.Created
+	newExtension.Updated = time.Now()
+
+	for _, newExtService := range newExtension.ListServices() {
+		_, err := extension.GetService(newExtService.ID)
+		if err != nil {
+			// If the service is new, set the creation time
+			newExtService.SetCreated(newExtension.Updated)
+		}
+	}
```
Comment (stefannica) [inline pkg/core/store/badger/extension.go:91]: I could be wrong, but I thought there was a domain.Extension method that would do this for you, so you wouldn't have to duplicate it here.

**Review by stefannica — APPROVED** [review]
One minor comment, otherwise looks great. It's amazing how much simpler the store implementation becomes with most of the logic moved into the domain.

File: pkg/core/store/badger/extension.go (hunk 1 above)
Comment (jsuchome) [reply pkg/core/store/badger/extension.go:56]: Well, the question is, if this is correct behavior (returning all items when query does not match anything)

File: pkg/core/store/badger/extension.go (hunk 2 above)
Comment (jsuchome) [reply pkg/core/store/badger/extension.go:28]: Great

File: pkg/core/store/badger/extension.go (hunk 1 above)
Comment (flaviodsr) [reply pkg/core/store/badger/extension.go:56]: Well in the case that it fails "all items" will be an empty list, so I would say yes.

Comment (jsuchome) [reply pkg/core/store/badger/extension.go:56]: OK, right, empty list, not all items. The question remains the same.

But I guess that's how we do it with other resources too.

Comment (flaviodsr) [reply pkg/core/store/badger/extension.go:56]: yes, I think that we agreed on that.

**Pull request merged**